assets/helpers/user/user.js

```diff
@@ -27,7 +27,7 @@ const init = (dispatch: Function) => {
 
   const uatMode = window.guardian && window.guardian.uatMode;
 
-  const emailFromSession: ?string = getSession('gu.email');
+  const emailFromPage: ?string = window.guardian.email || getSession('gu.email');
```

Comment:
The same practice from the line above, i.e.

```js
(window.guardian && window.guardian.email) || getSession('gu.email')
```

would be good in case `window.guardian` isn't defined. However, at this point I think it might be better when wrapped in a function for clarity:

```js
function getEmailFromPage(): ?string {
  if (window.guardian && window.guardian.email) {
    return window.guardian.email;
  }

  return getSession('gu.email');
}
```

Then line 67 becomes:

```js
} else if (getEmailFromPage()) {
```

Reply:
👍